cms/models.py

```diff
@@ -303,12 +305,23 @@ class FrequentlyAskedQuestion(Orderable):
     faqs_page = ParentalKey(CategorizedFaqsPage, related_name='faqs', null=True)
     question = models.TextField()
     answer = RichTextField()
+    slug = models.SlugField(unique=True, default=None, blank=True)
+
+    def save(self, *args, **kwargs):
+        if not self.slug:
+            for x in itertools.count(1):
+                self.slug = orig = slugify(self.question)
+                if not FrequentlyAskedQuestion.objects.filter(slug=self.slug).exists():
```

Comment:
But how will it tell if the slug with the number causes a collision? Since `self.slug` is defined in the previous line it will never contain the slug with the number, just the slug itself